kyaml/yaml/fns.go

```diff
@@ -123,6 +124,41 @@ func (e ElementSetter) Filter(rn *RNode) (*RNode, error) {
 	return NewRNode(e.Element), nil
 }
 
+// GetElementByIndex will return a Filter which can be applied to a sequence
+// node to get the element specified by the index
+func GetElementByIndex(index string) ElementPicker {
+	return ElementPicker{Index: index}
+}
+
+// ElementPicker picks the element with a specified index. Index starts from
+// 0 to len(list) - 1. a hyphen ("-") means the last index.
+type ElementPicker struct {
+	Index string
+}
+
+// Filter implements Filter
+func (p ElementPicker) Filter(rn *RNode) (*RNode, error) {
+	lastElement := false
+	idx, err := strconv.Atoi(p.Index)
```

Comment:
so all this parsing would be in the filter factory function instead of the filter